source/postprocess/melt_statistics.cc

```diff
@@ -0,0 +1,182 @@
+            // otherwise, set them to zero
+            std::vector<double> melt_fractions(n_q_points, 0.0);
+            if (const MaterialModel::MeltFractionModel<dim> *
+                melt_material_model = dynamic_cast <const MaterialModel::MeltFractionModel<dim>*> (&this->get_material_model()))
+              melt_material_model->melt_fractions(in, melt_fractions);
+
+            for (unsigned int q=0; q<n_q_points; ++q)
+              {
+                local_melt_integral += melt_fractions[q] * fe_values.JxW(q);
+                local_min_melt       = std::min(local_min_melt, melt_fractions[q]);
+                local_max_melt       = std::max(local_max_melt, melt_fractions[q]);
+              }
+
+          }
+
+      const double global_melt_integral
+        = Utilities::MPI::sum (local_melt_integral, this->get_mpi_communicator());
+      double global_min_melt = 0;
+      double global_max_melt = 0;
+
+      // now do the reductions that are
+      // min/max operations. do them in
+      // one communication by multiplying
+      // one value by -1
+      {
+        double local_values[2] = { -local_min_melt, local_max_melt };
+        double global_values[2];
+
+        Utilities::MPI::max (local_values, this->get_mpi_communicator(), global_values);
+
+        global_min_melt = -global_values[0];
+        global_max_melt = global_values[1];
+      }
+
+
+      // finally produce something for the statistics file
+      statistics.add_value ("Minimal melt fraction",
+                            global_min_melt);
+      statistics.add_value ("Total melt fraction",
+                            global_melt_integral);
+      statistics.add_value ("Maximal melt fraction",
+                            global_max_melt);
+
+      // also make sure that the other columns filled by this object
+      // all show up with sufficient accuracy and in scientific notation
+      {
+        const char *columns[] = { "Minimal melt fraction",
+                                  "Total melt fraction",
+                                  "Maximal melt fraction"
+                                };
+        for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+          {
+            statistics.set_precision (columns[i], 8);
+            statistics.set_scientific (columns[i], true);
+          }
+      }
+
+      output << global_min_melt << ", "
+             << global_melt_integral << ", "
+             << global_max_melt;
+
+      return std::pair<std::string, std::string> ("Melt fraction min/total/max:",
+                                                  output.str());
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(MeltStatistics,
+                                  "melt statistics",
+                                  "A postprocessor that computes some statistics about "
+                                  "the melt fraction, averaged by volume. ")
```

Comment:
Maybe add a sentence that it will output 0 if the material model does not provide melt fractions?